packages/runtime/src/useSSRComputation_Client.ts

```diff
@@ -0,0 +1,150 @@
+      }
+    }
+    const getCurrentResult = () => this.state.currentResult!;
+    this.subscription = this.state.module?.subscribe?.(getCurrentResult, updateResult, ...this.dependencies);
+  }
+
+  getResultIfModuleLoaded = () => {
+    const fn = this.state.module?.compute;
+    return fn ? fn(...this.dependencies) : NoResult;
+  }
+
+  loadAndRun = () => {
+    if (this.isDisposed) return;
+    this.state.importFn().then(module => {
+      if (this.isDisposed) return;
+      if (!module.compute) throw new Error('The SSR Computation module must have a compute function');
+      this.state.module = module;
+      this.handleSubscriptionIfModuleLoaded({ recomputeTheResult: true });
+    });
+  }
+
+  dispose = () => {
+    if (this.isDisposed) return;
+    this.isDisposed = true;
+    this.subscription?.unsubscribe();
+    this.subscription = undefined;
+  }
+}
+
+const useSSRComputation_Client = <TResult>(
+  importFn: ClientComputationFunction<TResult>,
+  dependencies: Dependency[],
+  options: Options,
+  relativePathToCwd: string
+): TResult | null => {
+  const [, forceUpdate] = useState(0);
+  const clientState = useRef<ClientState<TResult>>({
+    importFn,
+    currentResult: null,
+  });
+
+  const cache = getSSRCache();
+  const parsedDependencies = parseDependencies(dependencies);
+  const skip = !!options.skip;
+  // relativePathToCwd is used to make sure that the cache key is unique for each module
+  // and it's not affected by the file that calls it
+  const cacheKey = calculateCacheKey(relativePathToCwd, parsedDependencies);
+
+  const updateResult = useCallback((newResult: TResult, rerender: boolean) => {
+    cache[cacheKey] = {
+      result: newResult,
+      isSubscription: cache[cacheKey]?.isSubscription || !!clientState.current?.module?.subscribe,
+    }
+    clientState.current.currentResult = newResult;
+    if (rerender) {
+      forceUpdate(prevState => prevState + 1);
+    }
+  }, [cacheKey]);
+
+  const executor = useMemo(() => {
+    clientState.current.currentExecutor?.dispose();
+    clientState.current.currentExecutor = undefined;
+
+    if (skip) return null;
+    const updateResultAndRerender = (newResult: TResult) => updateResult(newResult, true);
+    clientState.current.currentExecutor = new ComputationExecutor(parsedDependencies, updateResultAndRerender, clientState.current);
+
+    const computedResult = clientState.current.currentExecutor.getResultIfModuleLoaded();
+    if (computedResult !== NoResult) {
+      updateResult(computedResult, false);
+    }
+    return clientState.current.currentExecutor;
+  }, [skip, cacheKey, updateResult]);
+
+  useEffect(() => {
+    // If the module is loaded, the current is result is calculated at the same render.
+    // And here we need to subscribe to the module if it has a subscribe function.
+    executor?.handleSubscriptionIfModuleLoaded({ recomputeTheResult: false });
+
+    return () => {
+      executor?.dispose();
```

Comment:
The executor is disposed here as well to handle the case when the component is completely unmounted. In this case, the `useMemo` function will not be executed.